From 67b3f0f3c333e771f5b3455b9b690082985614db Mon Sep 17 00:00:00 2001
From: James E Keenan <jkeenan@cpan.org>
Date: Thu, 11 May 2017 04:23:40 -0400
Subject: Correct the order of tests of chmod(). (#294)

Per code review by haarg, the order of tests was wrong in the first place.
Hence, correctly re-ordering them is a better repair than changing one test's
description.

For: https://github.com/Perl-Toolchain-Gang/ExtUtils-MakeMaker/pull/294

[Debian note: this is a prerequisite for the CVE-2017-6512 fix in
File-Path]

Bug: https://github.com/Perl-Toolchain-Gang/ExtUtils-MakeMaker/pull/294
Patch-Name: fixes/extutils_file_path_compat.diff
---
 cpan/ExtUtils-MakeMaker/t/eu_command.t | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/cpan/ExtUtils-MakeMaker/t/eu_command.t b/cpan/ExtUtils-MakeMaker/t/eu_command.t
index 269aa5c..32a6f59 100644
--- a/cpan/ExtUtils-MakeMaker/t/eu_command.t
+++ b/cpan/ExtUtils-MakeMaker/t/eu_command.t
@@ -151,20 +151,21 @@ BEGIN {
         is( ((stat('testdir'))[2] & 07777) & 0700,
             0100, 'change a dir to execute-only' );
 
-        # change a dir to read-only
-        @ARGV = ( '0400', 'testdir' );
+        # change a dir to write-only
+        @ARGV = ( '0200', 'testdir' );
         ExtUtils::Command::chmod();
 
         is( ((stat('testdir'))[2] & 07777) & 0700,
-            0400, 'change a dir to read-only' );
+            0200, 'change a dir to write-only' );
 
-        # change a dir to write-only
-        @ARGV = ( '0200', 'testdir' );
+        # change a dir to read-only
+        @ARGV = ( '0400', 'testdir' );
         ExtUtils::Command::chmod();
 
         is( ((stat('testdir'))[2] & 07777) & 0700,
-            0200, 'change a dir to write-only' );
+            0400, 'change a dir to read-only' );
 
+        # remove the dir we've been playing with
         @ARGV = ('testdir');
         rm_rf;
         ok( ! -e 'testdir', 'rm_rf can delete a read-only dir' );
